apps/common/lib/lexical/ast.ex

```diff
@@ -55,6 +55,50 @@ defmodule Lexical.Ast do
     end
   end
 
+  @doc """
+  Traverses the given ast until the given position.
+  """
+  def traverse_until(ast, acc, prewalk_fn, %Position{} = position) do
```

Comment:
The first time I saw this function, it wasn't clear to me that it would only turn the accumulator, since `traverse*` functions usually return `{thing_being_traversed, acc}`. Adding a spec would help, but maybe it's also worth calling this `walk_until`? Just to break the expectation compared to other traversal functions.

Reply:
yeah, but it's using `Zipper.traverse` `walk` is pretty generic, the elixir functions specify `prewalk` or `postwalk`. 

IIRC you made a suggestion earlier to only return the accumulator, prior I was returning both the element and the accumulator.

Reply:
> IIRC you made a suggestion earlier to only return the accumulator, prior I was returning both the element and the accumulator.

How dare you make this about _me_!

😉 You're almost right though, but my earlier suggestion was to change the signature of the passed function since we were ignoring the returned node, e.g.  `{_node, acc} = fun.(node)`. Maybe instead we should not ignore it and update the zipper using `Zipper.update/2`.